woocommerce/woocommerce
# Fix remote inbox notification showing up for wrong country

### All Submissions:






### Changes proposed in this Pull Request:

Makes sure we don't trigger the Remote inbox notifications until either the onboarding wizard has been skipped or finished, or when the user has updated the store address. This was suggested as a solution here: pdibGW-LT-p2#comment-590 

Closes [#34272](https://github.com/woocommerce/woocommerce/issues/34272) .

### How to test the changes in this Pull Request:

1. Load this branch and start a brand new store
2. add this piece of code to your mu-plugin, if you don't have an mu-plugin set up you could install the [code snippets](https://wordpress.org/plugins/code-snippets/) plugin and add this as part of that:
```
function add_datasource_remote_inbox_notifications( $data_sources, $id ) {
	if ( "remote_inbox_notifications" === $id ) {
		$data_sources[] = "https://gist.githubusercontent.com/louwie17/bd6f9fb6e0dec763f3dd9ed204924689/raw/36a64c663691ae495883a7c837a0c2bc231a3a9e/inbox-notifications.json";
	}
	return $data_sources;
}
add_filter( 'data_source_poller_data_sources', 'add_datasource_remote_inbox_notifications', 10, 2 );
```
3. Without finishing the onboarding wizard, go to **WooCommerce > Products > All products** and click Activity in the top right.
4. You should not see the **Navigate to add products only in USA and Canada** note
5. Now start the onboarding by selecting South Africa as the store location and finish the rest of the wizard.
6. Once on the home screen more notes should show, but not one with the title **Navigate to add products only in USA and Canada**, see the [inbox notification json](https://gist.githubusercontent.com/louwie17/bd6f9fb6e0dec763f3dd9ed204924689/raw/36a64c663691ae495883a7c837a0c2bc231a3a9e/inbox-notifications.json) for the full rules (should be added as part of the above snippet. You should see **Navigate to onboarding wizard** for example.
7. Start another brand new store with the same snippet above. Finish the onboarding, but this time selecting an American or Canadian address. Once you are back on the home screen it should show the **Navigate to add products only in USA and Canada** note.

### Other information:




### FOR PR REVIEWER ONLY:
